scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+        of the objective function.
+
+
+    options : dict, optional
+        A dictionary of solver options. All methods accept the following
+        options:
+
+            n_init : int, positive (default = 1)
+               Number of random initializations of the starting
+               permutation matrix that the FAQ algorithm will undergo.
+               n_init automatically set to 1 if init_method = 'barycenter'
+            init_method : string (default = 'barycenter')
+               The initial position chosen:
+               "barycenter" : the non-informative "flat doubly stochastic
+               matrix,":math:'J=1*1^T /n' , i.e the barycenter of the
+               feasible region
+               "rand" : some random point near :math:'J, (J+K)/2', where K
+               is some random doubly stochastic matrix
+            max_iter : int, positive (default = 30)
+               Integer specifying the max number of Franke-Wolfe iterations.
+               FAQ typically converges with modest number of iterations.
+            shuffle_input : bool (default = True)
+               Gives users the option to shuffle the nodes of 'cost_matrix'
+               to avoid results from inputs that were already matched.
+            eps : float (default = 0.1)
+               A positive, threshold stopping criteria such that FW
+               continues to iterate while Frobenius norm of
+               :math:'(P_{i}-P_{i+1}) > eps'
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+
+            col_ind : 1-D array
+                An array of column indices corresponding to the optimal optimal
+                permutation (with the fixed seeds given) on the
+                nodes of B, to best minimize the objective function
+                :math:'f(P) = trace(A^T PBP^T )'.
+            score : float
+                The optimal value of the objective function.
+            nit : int
+                The total number of iterations performed in all phases.
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015.
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, Seeded graph matching, Pattern Recognit. 87 (2019)
+           203–215
+
+    Examples
+    --------
+
+    >>> cost = [[0,22,53,53],[22,0,40,62],[53,40,0,55],[53,62,55,0]]
+    >>> dist = [[0,3,0,2],[3,0,0,1],[0,0,0,4],[2,1,4,0]]
+    >>> from scipy.optimize import quadratic_assignment
+    >>> quadratic_assignment(cost,dist)
+       col: array([2, 3, 0, 1])
+       nit: 30
+       row: array([0, 1, 2, 3])
+     score: 790
+
```

Comment:
It would be helpful to see how the score is calculated from `col_ind` in the example - maybe show that the value calculated from `col_ind` is equal to `res.score`.